lib/ai/chat.go

```diff
@@ -82,69 +82,131 @@ func labelsToPbLabels(vals []*assistantservice.Label) []Label {
 
 // Summary create a short summary for the given input.
 func (chat *Chat) Summary(ctx context.Context, message string) (string, error) {
-	var opts []grpc.DialOption
+	resp, err := chat.client.svc.CreateChatCompletion(
+		ctx,
+		openai.ChatCompletionRequest{
+			Model: openai.GPT4,
+			Messages: []openai.ChatCompletionMessage{
+				{Role: openai.ChatMessageRoleSystem, Content: promptSummarizeTitle},
+				{Role: openai.ChatMessageRoleUser, Content: message},
+			},
+		},
+	)
 
-	opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
-
-	conn, err := grpc.Dial(chat.client.apiURL, opts...)
 	if err != nil {
 		return "", trace.Wrap(err)
 	}
-	defer conn.Close()
-
-	assistantClient := assistantservice.NewAssistantServiceClient(conn)
 
-	resp, err := assistantClient.TitleSummary(ctx, &assistantservice.TitleSummaryRequest{
-		Message: message,
-	})
-
-	if err != nil {
-		return "", trace.Wrap(err)
-	}
+	return resp.Choices[0].Message.Content, nil
+}
 
-	return resp.Title, nil
+type StreamingMessage struct {
+	Role   string
+	Idx    int
+	Chunks <-chan string
+	Error  <-chan error
```

Comment:
Be careful, exported channels without any godocs as to the correct way to read from them will get complicated quickly.

Reply:
Yep, I've added godocs in relevant places now. Hadn't gotten around to this yet signified by the PR's draft state.